server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -2783,84 +2786,140 @@ public ServiceOffering updateServiceOffering(final UpdateServiceOfferingCmd cmd)
-        // else if (!tags.trim().isEmpty() && offeringHandle.getTags() != null)
-        // {
-        // //new tags + existing tags
-        // List<String> oldTags = csvTagsToList(offeringHandle.getTags());
-        // List<String> newTags = csvTagsToList(tags);
-        // oldTags.addAll(newTags);
-        // offering.setTagsArray(oldTags);
-        // }
-        // else if(!tags.trim().isEmpty())
-        // {
-        // //new tags; NO existing tags
-        // offering.setTagsArray(csvTagsToList(tags));
-        // }
-        // }
-
-        if (updateNeeded && !_serviceOfferingDao.update(id, offering)) {
-            return null;
+        if (tags != null) {
+            fields.add("tags: " + tags);
+            offering.setTags(StringUtils.cleanupTags(tags));
         }
-        List<ServiceOfferingDetailsVO> detailsVO = new ArrayList<>();
-        if(detailsUpdateNeeded) {
-            SearchBuilder<ServiceOfferingDetailsVO> sb = _serviceOfferingDetailsDao.createSearchBuilder();
-            sb.and("offeringId", sb.entity().getResourceId(), SearchCriteria.Op.EQ);
-            sb.and("detailName", sb.entity().getName(), SearchCriteria.Op.EQ);
-            sb.done();
-            SearchCriteria<ServiceOfferingDetailsVO> sc = sb.create();
-            sc.setParameters("offeringId", String.valueOf(id));
-            if(!filteredDomainIds.equals(existingDomainIds)) {
-                sc.setParameters("detailName", ApiConstants.DOMAIN_ID);
-                _serviceOfferingDetailsDao.remove(sc);
-                for (Long domainId : filteredDomainIds) {
-                    detailsVO.add(new ServiceOfferingDetailsVO(id, ApiConstants.DOMAIN_ID, String.valueOf(domainId), false));
-                }
-            }
-            if(!filteredZoneIds.equals(existingZoneIds)) {
-                sc.setParameters("detailName", ApiConstants.ZONE_ID);
-                _serviceOfferingDetailsDao.remove(sc);
-                for (Long zoneId : filteredZoneIds) {
-                    detailsVO.add(new ServiceOfferingDetailsVO(id, ApiConstants.ZONE_ID, String.valueOf(zoneId), false));
-                }
+
+        if (hostTag != null) {
+            if (hostTag.trim().isEmpty()) {
+                fields.add("removing host tag");
+                offering.setHostTag(null);
+            } else {
+                fields.add("host tag: " + hostTag);
+                offering.setHostTag(hostTag);
             }
         }
-        if (!detailsVO.isEmpty()) {
-            for (ServiceOfferingDetailsVO detailVO : detailsVO) {
-                _serviceOfferingDetailsDao.persist(detailVO);
-            }
+
+        s_logger.debug(String.format("Updating service offering ID [%s] with [%s].", serviceOfferingId, String.join(", ", fields)));
+
+        if (!_serviceOfferingDao.update(serviceOfferingId, offering)) {
+            s_logger.warn(String.format("Can't update service offering ID [%s] with [%s].", serviceOfferingId, String.join(", ", fields)));
+        }
```

Comment:
I think these contain the same info for the operator;
```suggestion

        if (_serviceOfferingDao.update(serviceOfferingId, offering)) {
            s_logger.debug(String.format("Updated service offering ID [%s] with [%s].", serviceOfferingId, String.join(", ", fields)));
        } else {
            s_logger.warn(String.format("Can't update service offering ID [%s] with [%s].", serviceOfferingId, String.join(", ", fields)));
        }
```